lkmpg.tex

```diff
@@ -1689,6 +1689,21 @@ \subsection{Spinlocks}
 
 \samplec{examples/example_spinlock.c}
 
+Taking 100\% of a CPU's resources comes with greater responsibility.
+Situations where the kernel code monopolizes a CPU are called \textbf{atomic contexts}.
+Holding a spinlock is one of those situations.
+Sleeping in atomic contexts may leave the system hanging, as the occupied CPU devotes 100\% of its resources doing nothing but sleeping.
+In some worse cases the system may crash.
+Thus, sleeping in atomic contexts is considered a bug in the kernel.
+They are sometimes called "sleep-in-atomic-context" in some materials.
```

Comment:
Be aware of quotation marks in LaTex. See https://www.maths.tcd.ie/~dwilkins/LaTeXPrimer/QuotDash.html